### ISSUE TITLE
[Bug][configen] Generates unquoted str defaults when type hints include Optional[  ]

### ISSUE THREAD

**Issue by romesco** [facebookresearch/hydra#1174]
# 🐛 Bug
Almost identical issue to: #1071 except on a new edge case.
Configen will create fields typed with `Optional[str]`, but if there's a default value which is a str type, it will not include it in quotes. We should also test other supported Type hint permutations.

Ex:
Generates the field 
`weights_summary: Optional[str] = top` 
instead of the desired:
 `weights_summary: Optional[str] = "top"`

## To reproduce
```
git clone https://github.com/romesco/hydra-lightning
configen --config-dir=configen/conf
```

Will create PR similar to #1072 for fix.

## System information
- **Hydra Version** :  `configen==0.9.0dev4`
- **Python version** : 3.8.6
- **Virtual environment type and version** : pyenv-virtualenv 1.2.20
- **Operating system** : Ubuntu 18.04

**Comment by omry** [comment]
ping, still planning on fixing it?

**Comment by romesco** [comment]
Yep! Thanks for the ping. It wasn't in the critical path, but it is important!

**Comment by romesco** [comment]
woah, return of the legend

**Comment by omry** [comment]
Come back to the Hydra chat. it's desolate ;)

**Comment by omry** [comment]
I reproduced this against the current Configen code using `weights_summary: Optional[str] = "top"`. It now generates:

```python
from dataclasses import dataclass, field
from typing import Optional


@dataclass
class ExampleConf:
    _target_: str = "configen_1174_example.Example"
    weights_summary: Optional[str] = "top"
```

The report used `hydra-configen==0.9.0.dev4`, which was published before https://github.com/facebookresearch/hydra/pull/1072 added quoting based on the runtime default type. The current implementation therefore already handles this case correctly. Closing as fixed.
